azurerm/internal/services/network/virtual_network_gateway_resource.go

```diff
@@ -396,6 +410,24 @@ func resourceArmVirtualNetworkGatewayRead(d *schema.ResourceData, meta interface
 			return fmt.Errorf("Error setting `ip_configuration`: %+v", err)
 		}
 
+		futurePackage, err := client.GetVpnProfilePackageURL(ctx, resGroup, name)
+		if err != nil {
+			return fmt.Errorf("Error generating VPN Profile Package for AzureRM Virtual Network Gateway %q (Resource Group %q): %+v", name, resGroup, err)
+		}
+
+		if err = futurePackage.WaitForCompletionRef(ctx, client.Client); err != nil {
+			return fmt.Errorf("Error waiting for completion of generating VPN Profile Package for AzureRM Virtual Network Gateway %q (Resource Group %q): %+v", name, resGroup, err)
+		}
+
+		respPackage, err := futurePackage.Result(*client)
+		if err != nil {
+			return fmt.Errorf("Error retrieving VPN Profile Package URL for AzureRM Virtual Network Gateway %q (Resource Group %q): %+v", name, resGroup, err)
+		}
```

Comment:
Can I use a future like this? It's not done anywhere else, what would the best solution be?

Reply:
It should work like that, issue on similar method has been marked as a bug.